google/api_core/exceptions.py

```diff
@@ -433,13 +472,21 @@ def from_http_response(response):
     errors = payload.get("error", {}).get("errors", ())
     # In JSON, details are already formatted in developer-friendly way.
     details = payload.get("error", {}).get("details", ())
+    error_info_type = "type.googleapis.com/google.rpc.ErrorInfo"
+    error_info = [d for d in details if d.get("@type", "") == error_info_type]
```

Comment:
The `pytype` failure here is puzzling.  Perhaps we need to add a type assertion for either `payload` or `details`?

Reply:
I refactored this code.